ambry-frontend/src/main/java/com/github/ambry/frontend/s3/S3MultipartUploadPartHandler.java

```diff
@@ -80,13 +213,88 @@ void handle(RestRequest restRequest, RestResponseChannel restResponseChannel,
         // Set S3 ETag header
         String blobId = (String) restResponseChannel.getHeader(LOCATION);
         restResponseChannel.setHeader("ETag", blobId);
-        //TODO [S3]: remove x-ambry- headers
-        callback.onCompletion(null, null);
-      } catch (RestServiceException e) {
-        callback.onCompletion(null, e);
+        finalCallback.onCompletion(null, null);
+      }, restRequest.getUri(), logger, finalCallback);
+    }
+
+    /**
+     * Parse {@link BlobInfo} from the request arguments. This method will also ensure that the correct account and
+     * container objects are attached to the request.
+     * @return the {@link BlobInfo} parsed from the request arguments.
+     * @throws RestServiceException if there is an error while parsing the {@link BlobInfo} arguments.
+     */
+    private BlobInfo getBlobInfoFromRequest() throws RestServiceException {
+      // TODO [S3] may consolidate and move it to a central place.
+      long propsBuildStartTime = System.currentTimeMillis();
+      accountAndContainerInjector.injectAccountContainerForNamedBlob(restRequest, frontendMetrics.putBlobMetricsGroup);
+      if (RestUtils.isDatasetVersionQueryEnabled(restRequest.getArgs())) {
+        accountAndContainerInjector.injectDatasetForNamedBlob(restRequest);
+      }
```

Comment:
Since it is specific to dataset code, it can be removed?

Reply:
You are right. Removed.